File: 1642-water-bottles/water-bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        ans = numBottles
        bot, exc = numBottles, numExchange

        while bot:
            x = bot // exc
            y = bot - (exc * x)
            ans += x
            bot //= exc
            bot += y

            if bot < exc:
                return ans

File: 1642-water-bottles/water-bottles.js
/**
 * @param {number} numBottles
 * @param {number} numExchange
 * @return {number}
 */
var numWaterBottles = function(numBottles, numExchange) {
    let div = Math.floor((numBottles - 1) / (numExchange - 1));
    let ans = numBottles + div;

    return ans;
};